lib/safer_rails_console.rb

```diff
@@ -54,8 +58,60 @@ class Configuration
       config_accessor(name) { value }
     end
 
+    CONFIG_ENV_DEFAULTS = {
+      short_name: {
+        config: config[:environment_names][ENVIRONMENT_NAME],
+        env_var: 'SAFER_RAILS_CONSOLE_SHORT_NAME',
+        type: 'string'
+      },
+      prompt_color: {
+        config: config[:environment_prompt_colors],
+        env_var: 'SAFER_RAILS_CONSOLE_PROMPT_COLOR',
+        type: 'color'
+      },
+      sandbox_environment: {
+        config: config[:sandbox_environments],
+        env_var: 'SAFER_RAILS_CONSOLE_SANDBOX_ENVIRONMENT',
+        type: 'boolean'
+      },
+      warn_environment: {
+        config: config[:warn_environments],
+        env_var: 'SAFER_RAILS_CONSOLE_WARN_ENVIRONMENT',
+        type: 'boolean'
+      },
+      warn_text: {
+        config: config[:warn_text],
+        env_var: 'SAFER_RAILS_CONSOLE_WARN_TEXT',
+        type: 'string'
+      }
+    }.freeze
+
     def set(**new_config)
+      CONFIG_ENV_DEFAULTS.each do |_, values|
```

Comment:
The end result of this is that the DEFAULT_CONFIG is modified in place prior to the new settings (`new_config`) passed into this method by, under normal circumstances, the Railtie. A couple concerns I have with this are:

1. We're mutating part of a constant, that is supposedly frozen. Now, this is Ruby, so nothing is _really_ constant, but this might be surprising behavior to someone working on this bit of code later.
2. The precedence order of configuration is `Rails.configuration.safer_rails_console` (via railtie) > `SAFER_RAILS_CONSOLE` ENVs > Actual defaults.

Is \#2 really the precedence order we want, or do we want ENV to override _any_ configuration, either by hard-coded defaults or `Rails.configuration.safer_rails_console` modifications? If we _do_ want ENV to be a final override, putting these directly in the methods on lines 9-23 would be a really straightforward implementation, for example:

```
    def sandbox_environment?
      config.sandbox_environments.include?(::Rails.env.downcase)
    end
```

Could become something like:

```
def sandbox_environment?
  if ENV.key?('SAFER_RAILS_CONSOLE_SANDBOX_ENVIRONMENT')
    ActiveModel::Type::Boolean.new.cast(ENV['SAFER_RAILS_CONSOLE_SANDBOX_ENVIRONMENT'])
  else
    config.sandbox_environments.include?(::Rails.env.downcase)
  end
end
```

Essentially if the ENV is set, we don't even consult the OrderedHash in the Configurable.

Reply:
I agree that it may make more sense to have the order of precedence be ENVs > Rails config > defaults. I think it makes it more clear as to the intention of what the ENV variables are supposed to provide, which is a temporary override of configuration settings when instantiating a safer rails console. 

I really preferred your approach for effectively handling the ENV var support by pushing it out of the config layer. My only concern is that for things like the `prompt_color` can fail if the color code isn't found. However, I think we can add a simple check for its existence and pass `NONE` if it isn't found.